simperby/src/lib.rs

```diff
@@ -29,49 +31,65 @@ pub trait BlockExecutor {
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub enum StateTransition {
-    InsertValidator {
+    AddValidator {
         /// The public key of the validator.
         public_key: PublicKey,
         /// The weight of the validator.
-        weight: u64,
+        voting_power: VotingPower,
+    },
+    RemoveValidator {
+        /// The public key of the validator.
+        public_key: PublicKey,
     },
-    RemoveValidator(PublicKey),
-    /// This is accepted only if the signer of the transaction equals to the `delegator`.
     Delegate {
         /// The public key of the validator who delegates its voting right.
         delegator: PublicKey,
         /// The public key of the validator who is being delegated.
         delegatee: PublicKey,
         /// The target height of the block of this transaction, which is for preventing the replay attack.
-        target_height: u64,
+        target_height: BlockHeight,
+        /// The signature of (delegatee, target_height) signed by the delegator.
+        ///
+        /// This is a (non-essential) safety guard that assures that the delegator actually wants it.
+        /// Otherwise (if this field doesn't exist), any sincere block proposer would include the commitment proof
+        /// in the transaction payload **anyway**, to convince other honest validators.
+        ///
+        /// That's why we just provide a reserved field for that.
+        commitment: TypedSignature<(PublicKey, BlockHeight)>,
     },
-    /// This is accepted only if the signer of the transaction equals to the `delegator`.
     Undelegate {
         /// The public key of the validator who claims its voting right.
         delegator: PublicKey,
-        /// The target height of the block of this transaction, which is for preventing the replay attack.
-        target_height: u64,
-    },
-    InsertData {
-        key: String,
-        value: Vec<u8>,
     },
+    /// Records a new data.
+    InsertData { key: String, value: Vec<u8> },
+    /// Removes an existing data.
     RemoveData(String),
+    /// Updates the protocol version
+    UpdateVersion { version: String },
 }
 
+/// A unit of state transition of the blockchain.
+///
+/// Unlike other 'ordinary' blockchains, Simperby doesn't have a signer nor a signature.
```

Comment:
```suggestion
/// Unlike other "ordinary" blockchains, Simperby doesn't have a signer nor a signature.
```
For emphasizing, it's proper to use double quotation marks than single quotation marks.

Reply:
fixed